app/controllers/users/two_factor_authentication_setup_controller.rb

```diff
@@ -73,12 +74,7 @@ def two_factor_options_presenter
 
     def process_valid_form
       user_session[:mfa_selections] = @two_factor_options_form.selection
-
-      if user_session[:mfa_selections].first.present?
-        redirect_to confirmation_path(user_session[:mfa_selections].first)
-      else
-        redirect_to after_mfa_setup_path
-      end
+      redirect_to(first_mfa_selection_path || after_mfa_setup_path)
```

Comment:
Will `first_mfa_selection_path` ever return `nil`?

Do we need a `return if user_session[:mfa_selections].blank?` early return in that method?

Reply:
And just to clarify since I'm struggling to recall the relation of these changes, is this just to update existing references to the session value to match the new abstraction we've introduced here?

Reply:
It may be nil if a user chooses to skip the second MFA selection page or if theyre forced to be there again, I dont know if that still follows today.

Reply:
Hmm I see, I think I was confused by `confirmation_path` thinking it was a Rails URL helper that would always produce a string value and never `nil`. We might want some test coverage for those concern methods to confirm but I think I see now how we're expecting it to be `nil`.